server/test/services/apikey/ApiKeyServiceTest.java

```diff
@@ -0,0 +1,213 @@
+package services.apikey;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static play.test.Helpers.fakeRequest;
+
+import auth.ApiKeyGrants.Permission;
+import auth.CiviFormProfile;
+import auth.CiviFormProfileData;
+import auth.ProfileFactory;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import models.ApiKey;
+import org.junit.Before;
+import org.junit.Test;
+import play.data.DynamicForm;
+import play.data.FormFactory;
+import repository.ApiKeyRepository;
+import repository.ResetPostgres;
+import services.DateConverter;
+import services.apikey.ApiKeyService.ApiKeyCreationResult;
+import services.program.ProgramNotFoundException;
+
+public class ApiKeyServiceTest extends ResetPostgres {
+
+  ApiKeyService apiKeyService;
+  ApiKeyRepository apiKeyRepository;
+  ProfileFactory profileFactory;
+  CiviFormProfile adminProfile;
+  FormFactory formFactory;
+  DateConverter dateConverter;
+
+  @Before
+  public void setUp() throws Exception {
+    apiKeyService = instanceOf(ApiKeyService.class);
```

Comment:
general question: can these instanceOfs be in a @beforeClass ?

If so that'd be good and then have the other setup in a @Before.

Google style suggests having different/multiple @Before s for different setup concerns; not saying we have to do that, but made me wonder if these allow for that separation.

Reply:
`@BeforeClass` can only be used with static methods, so nothing that relies on dependency injection can use it